### DISCUSSION TITLE
Forecast with missing data

### DISCUSSION THREAD

**Discussion by mchessl** [ourownstory/neural_prophet#262] [Q&A - get help using NeuralProphet] [answered]
Hello,

I have a dataset that has 260 consecutive missing days in a total of 1700 days. From my understanding this should not be an issue as I am not using the AR-Net. My goal is to forecast the 260 missing days and the 30 days after the last date of the dataset.

However I got the message "More than 30 consecutive missing values encountered in column y. Please preprocess data manually".

I am using events. I have set up n_lags = 0 to ensure AR-Net is not used.

Do you know what I should do to be able to forecast such dataset ?

Thanks a lot for your help.
Kind regards

**Comment by ourownstory** [comment, marked as answer]
Hi @mchessl 
What you are trying should work, though it can still be tricky, as trend segments may be entirely in the missing section. (you can work around this by using few changepoints (<5) or regularizing trend).

However, you should not be getting an error. 
Do you mind providing the minimal code needed to reproduce the error, so we can debug it?

Thank you

@HansikaPH might this be related to events?

**Reply by mchessl** [reply]
Hi @ourownstory ,

I have just found the problem : I was fitting model with the missing dates and null values with those days.
I have removed those days in the fitted dataset, while providing them to the forecast and it did work.
Thanks again for your help,

Kind regards.

**Reply by ourownstory** [reply]
Hi @mchessl 

Good to hear!
If you were happy with the results, this might be a good case for an example notebook that might help other people, too.
If you would like to contribute a narrated notebook of what you tried to do and how you did it, I'd be happy to accept a pull request!

Example for inspiration: https://github.com/ourownstory/neural_prophet/blob/master/example_notebooks/season_multiplicative_air_travel.ipynb

**Reply by Erickurashi** [reply]
Hi, I still couldn't find the solution for how to solve this issue. Which part should I change? Thank you.

**Reply by mchessl** [reply]
Hi,

Sorry for my late reply @ourownstory, I won't be able to to such notebook I am at all used with that unfortunately.

@Erickurashi what helped me here was to removed from the dataset the various dates where I did not have a volume, instead of having those date with null values. In the meantime I did provide those days in the future dataframe to forecast, so that they were forecasted. Hope this helps you.

Kind regards
